### PR TITLE
Filterable List: Redesign and Updates

### PR DESCRIPTION
**Pull request by mistergone**
This PR contains the LESS changes necessary to implement a new design across the variety of Filterable List pages on our site.

## Additions
- Added code to hide dates for the Consumer Reporting Companies list
- Added code so that the "categories" element will not appear when there are no categories in a post preview entry.
- Added code so that the pipe ("|") separator appears within a `SPAN`, allowing it to be styled.
- Added LESS to use `grid` for better placement of the category and date (previously used `float`), and for rearrangement in mobile view.

## Testing
1. Pull down this branch and get it running.
2. Check out filterable list pages!

### List of Filterable List pages (assuming localhost):
- http://localhost:8000/administrative-adjudication-proceedings/administrative-adjudication-docket/
- http://localhost:8000/policy-compliance/notice-opportunities-comment/archive-closed/
- http://localhost:8000/about-us/blog/
- http://localhost:8000/data-research/cfpb-researchers/
- http://localhost:8000/policy-compliance/enforcement/actions/
- http://localhost:8000/policy-compliance/amicus/
- http://localhost:8000/policy-compliance/rulemaking/final-rules/
- http://localhost:8000/foia-requests/foia-electronic-reading-room/
- http://localhost:8000/policy-compliance/guidance/supervisory-guidance/
- http://localhost:8000/about-us/newsroom/
- http://localhost:8000/policy-compliance/notice-opportunities-comment/open-notices/
- http://localhost:8000/policy-compliance/enforcement/petitions/
- http://localhost:8000/data-research/research-reports/
- http://localhost:8000/policy-compliance/rulemaking/rules-under-development/
- http://localhost:8000/consumer-tools/credit-reports-and-scores/consumer-reporting-companies/companies-list/

## Screenshots
screen shot 2019-02-07 at 2 59 21 pm

screen shot 2019-02-07 at 3 02 30 pm

## Notes
- There is still some browser compatibility testing to do, so if you find something doesn't work in another browser, let me know!

## Todos
- Add linebreaks for multiple categories in mobile view
- Remove icons from FOIA Reports page

## Checklist
- [x] PR has an informative and human-readable title
- [x] Changes are limited to a single goal (no scope creep)
- [x] Code can be automatically merged (no conflicts)
- [ ] Code follows the standards laid out in the [CFPB development guidelines](https://github.com/cfpb/development)
- [ ] Passes all existing automated tests
- [ ] Any _change_ in functionality is tested
- [x] New functions are documented (with a description, list of inputs, and expected output)
- [x] Placeholder code is flagged / future todos are captured in comments
- [ ] Visually tested in supported browsers and devices (see checklist below :point_down:)
- [x] Project documentation has been updated
- [x] Reviewers requested with the [Reviewers tool](https://help.github.com/articles/requesting-a-pull-request-review/) :arrow_right:

### Browsers

- [x] Chrome
- [x] Firefox
- [x] Safari
- [x] Internet Explorer 8, 9, 10, and 11
- [x] Edge
- [x] iOS Safari
- [x] Chrome for Android

### Accessibility

- [x] Keyboard friendly
- [x] Screen reader friendly

### Other

- [x] Is useable without CSS
- [x] Is useable without JS
- [x] Flexible from small to large screens
- [x] No linting errors or warnings
- [x] JavaScript tests are passing

### CODE REVIEW

File: cfgov/unprocessed/css/organisms/post-preview.less [hunk 1]
```diff
@@ -83,16 +83,45 @@
...
+    .respond-to-max( @bp-xs-max, {
+        grid-template-areas:
+            "header"
+            "image"
+            "content";
+    } );
+    
     + .o-post-preview {
-        margin-top: unit( @grid_gutter-width * 2 / @base-font-size-px, em);
+        margin-top: unit( 30px / @base-font-size-px, em );
```
Comment (anselmbradford) [inline cfgov/unprocessed/css/organisms/post-preview.less]: Why this change when we use `@grid_gutter-width` [elsewhere in this PR](https://github.com/cfpb/cfgov-refresh/pull/4811/files#diff-f8dbb35d10b4d3ed1942d92505379790R175)? Are we moving away from that?

File: cfgov/unprocessed/css/organisms/post-preview.less [hunk 2]
```diff
@@ -83,16 +83,45 @@
...
 
 .o-post-preview {
+    border-bottom: 1px solid @meta-header_border;
+    padding-bottom: unit( 15px / @base-font-size-px, em );
+    display: grid;
+    grid-template-rows: auto;
+    grid-template-columns: auto;
+    grid-template-areas:
+        "header image"
+        "content image";
```
[the remark this answers is not in the document]
Comment (anselmbradford) [reply cfgov/unprocessed/css/organisms/post-preview.less]: Actually, are these used? I don't see references to these in `grid-area` values, unless I'm misunderstanding how the space works.

File: cfgov/jinja2/v1/_includes/organisms/post-preview.html [hunk 3]
```diff
@@ -77,6 +77,26 @@
     {% set show_categories = cat_controls.show_preview_categories if cat_controls is defined else true %}
     <article class="o-post-preview">
         <div class="m-meta-header">
+            {% if post.categories.exists() and show_categories %}
```
Comment (anselmbradford) [inline cfgov/jinja2/v1/_includes/organisms/post-preview.html]: TRIVIAL: If this is swapped around and `show_categories` is checked first it could bail out without needing to walk the object chain inside `post`.

File: cfgov/unprocessed/css/organisms/post-preview.less (hunk 2 above)
[the remark this answers is not in the document]
Comment (mistergone) [reply cfgov/unprocessed/css/organisms/post-preview.less]: Those aren't spaces in the header name, they are for the "image" area in the grid, but I now see an "image" area was actually not necessary in the final code, so I need to remove all references to it. :)

File: cfgov/jinja2/v1/_includes/organisms/post-preview.html [hunk 4]
```diff
@@ -77,6 +77,26 @@
...
+                    {{ category_slug.render(category='blog', href=url) }}
+                {% else %}
+                    {% for cat in post.categories.all() %}
+                        {% if loop.index > 1 %}
+                            <span class="m-meta-header_separator">|</span>
+                        {% endif %}
+                        {{ category_slug.render(category=cat.name, href=url) }}
+                    {% endfor %}
+                {% endif %}
+                {# pipe ("|") separator logic #}
```
Comment (anselmbradford) [inline cfgov/jinja2/v1/_includes/organisms/post-preview.html]: TRIVIAL: Capitalize `Pipe` as you do `Newsroom` in the prior comment. Also ending punctuation period wouldn't hurt.

File: cfgov/jinja2/v1/_includes/organisms/post-preview.html [hunk 5]
```diff
@@ -77,6 +77,26 @@
...
+                {% if show_date %}
```
Comment (anselmbradford) [inline cfgov/jinja2/v1/_includes/organisms/post-preview.html]: Would it make sense to move this into the if block above it and have `if … elif … else`? Or does that combine two unrelated things?

Comment (anselmbradford) [reply cfgov/jinja2/v1/_includes/organisms/post-preview.html]: Ooo I guess that wouldn't work if you needed this and the stuff in the else block, never mind.

File: cfgov/unprocessed/css/organisms/post-preview.less (hunk 1 above)
Comment (mistergone) [reply cfgov/unprocessed/css/organisms/post-preview.less]: This could change to `@grid_gutter-width` ( minus the `* 2`), so I can update that.

File: cfgov/unprocessed/css/organisms/post-preview.less [hunk 6]
```diff
@@ -102,20 +131,49 @@
                 vertical-align: bottom;
             }
         } );
+
+        &_left {
+            .u-webfont-medium();
```
Comment (anselmbradford) [inline cfgov/unprocessed/css/organisms/post-preview.less]: This mixin is [deprecated](https://github.com/cfpb/capital-framework/blob/4164814946955b5fb4bee48e02f085219cfcaba5/src/cf-core/src/cf-base.less#L9) and shouldn't be used.

File: cfgov/unprocessed/css/organisms/post-preview.less [hunk 7]
```diff
@@ -102,20 +131,49 @@
...
+            display: inline-block;
+            float: none;
+            font-size: 1.125em;
+            grid-area: category;
+            line-height: 1.33333;
```
Comment (anselmbradford) [inline cfgov/unprocessed/css/organisms/post-preview.less]: This value looks like the result of a calculated value. Was this copied from production code by chance?

File: cfgov/jinja2/v1/_includes/organisms/post-preview.html (hunk 5 above)
Comment (mistergone) [reply cfgov/jinja2/v1/_includes/organisms/post-preview.html]: They're unrelated.

The code above is for separating in the case of multiple categories. The `if show_date` is to prevent a Pipe character from rendering if categories are visible but date is hidden.

File: cfgov/unprocessed/css/organisms/post-preview.less [hunk 8]
```diff
@@ -102,20 +131,49 @@
...
+
+            .m-meta-header_separator {
+                padding-left: 0.25em;
+                padding-right: 0.25em;
+            }
+        }
+
+        &_right {
+            display: inline-block;
+            float: none;
```
Comment (anselmbradford) [inline cfgov/unprocessed/css/organisms/post-preview.less]: Possibly the shared settings could be combined into:

```less
&_left,
&_right {
    display: inline-block;
    float: none;
}
```

File: cfgov/unprocessed/css/organisms/post-preview.less [hunk 9]
```diff
@@ -102,20 +131,49 @@
...
+            grid-area: date;
+            line-height: 1.5;
+
+            .a-date {
+                line-height: 1.7143;
```
Comment (anselmbradford) [inline cfgov/unprocessed/css/organisms/post-preview.less]: This value looks suspect too. Magic number. How's this calculated?

File: cfgov/unprocessed/css/organisms/post-preview.less (hunk 7 above)
Comment (mistergone) [reply cfgov/unprocessed/css/organisms/post-preview.less]: No, it represents a `line-height` of 24px for a `font-size` of 18px. Is there a more preferable way to express it?

File: cfgov/unprocessed/css/organisms/post-preview.less (hunk 9 above)
Comment (mistergone) [reply cfgov/unprocessed/css/organisms/post-preview.less]: It's 24/14 - `line-height` of 24px, `font-size` of 14px. Is there a better/more preferred way for us to express that?

File: cfgov/unprocessed/css/organisms/post-preview.less (hunk 8 above)
Comment (mistergone) [reply cfgov/unprocessed/css/organisms/post-preview.less]: If that's preferable, sure!

File: cfgov/unprocessed/css/organisms/post-preview.less [hunk 10]
```diff
@@ -83,16 +83,45 @@
 */
 
 .o-post-preview {
+    border-bottom: 1px solid @meta-header_border;
+    padding-bottom: unit( 15px / @base-font-size-px, em );
```
Comment (anselmbradford) [inline cfgov/unprocessed/css/organisms/post-preview.less]: Looks like it should be `@grid_gutter-width / 2 / @base-font-size-px`

File: cfgov/unprocessed/css/organisms/post-preview.less (hunk 1 above)
Comment (anselmbradford) [reply cfgov/unprocessed/css/organisms/post-preview.less]: According to @Scotchester we should always use the variable.

File: cfgov/unprocessed/css/organisms/post-preview.less [hunk 11]
```diff
@@ -102,20 +131,49 @@
...
+
+            .a-date {
+                line-height: 1.7143;
+            }
+        }
+    }
+
+    &_content {
+        grid-area: content;
+        padding-right: unit( 30px / @base-font-size-px, em );
```
Comment (anselmbradford) [inline cfgov/unprocessed/css/organisms/post-preview.less]: Looked like it should use `@grid_gutter-width / @base-font-size-px`.

File: cfgov/unprocessed/css/organisms/post-preview.less (hunk 7 above)
Comment (anselmbradford) [reply cfgov/unprocessed/css/organisms/post-preview.less]: Prolly do the calculation `line-height: 24px / 18px`. I defer to @Scotchester for variables that would help here.

File: cfgov/jinja2/v1/_includes/organisms/post-preview.html (hunk 5 above)
Comment (anselmbradford) [reply cfgov/jinja2/v1/_includes/organisms/post-preview.html]: 👍

File: cfgov/unprocessed/css/organisms/post-preview.less (hunk 7 above)
Comment (Scotchester) [reply cfgov/unprocessed/css/organisms/post-preview.less]: You'd have to do `unit(24px / 18px)` to drop the unit.

Comment (Scotchester) [reply cfgov/unprocessed/css/organisms/post-preview.less]: But it'd probably be better to condense the `font-size`, `font-weight`, and `font-height` properties here into a call of [the `.heading-4()` mixin](https://github.com/cfpb/capital-framework/blob/canary/src/cf-core/src/cf-base.less#L120-L129).

File: cfgov/unprocessed/css/organisms/post-preview.less (hunk 2 above)
[the remark this answers is not in the document]
Comment (mistergone) [reply cfgov/unprocessed/css/organisms/post-preview.less]: Sorry, I was wrong, it does use `grid-area: image` in the code, I just missed it in my first pass-through! So the notation there, such as `"header image"`, should be correct...

File: cfgov/unprocessed/css/organisms/post-preview.less (hunk 9 above)
Comment (mistergone) [reply cfgov/unprocessed/css/organisms/post-preview.less]: Changing this to `unit(24px / 14px)`

**Comment by anselmbradford** [comment]
Do we care if the vertical alignment is inconsistent on the date?

FF:
screen shot 2019-02-12 at 12 55 39 pm

Chrome:
screen shot 2019-02-12 at 12 55 50 pm

File: cfgov/unprocessed/css/organisms/post-preview.less [hunk 12]
```diff
@@ -102,20 +131,50 @@
                 vertical-align: bottom;
             }
         } );
+
+        &_left,
+        &_right {
+            display: inline-block;
+            float: none;
```
Comment (anselmbradford) [inline cfgov/unprocessed/css/organisms/post-preview.less]: This float is overriding https://github.com/cfpb/capital-framework/blob/canary/src/cf-typography/src/molecules/meta-header.less#L9-L15
Do we know if this is used anywhere else or could the float be removed entirely from CF? If it could can we put a TODO here and/or create an issue in CF?

File: cfgov/unprocessed/css/organisms/post-preview.less (hunk 2 above)
[the remark this answers is not in the document]
Comment (anselmbradford) [reply cfgov/unprocessed/css/organisms/post-preview.less]: Ahh that's cool, I hadn't used `grid-template-areas` before, so it's like visually laying out the grid. Nice!

File: cfgov/jinja2/v1/_includes/organisms/post-preview.html [hunk 13]
```diff
@@ -77,6 +77,26 @@
     {% set show_categories = cat_controls.show_preview_categories if cat_controls is defined else true %}
     <article class="o-post-preview">
         <div class="m-meta-header">
+            {% if show_categories and post.categories.exists()  %}
+            <div class="m-meta-header_left">
+                {# Newsroom Blog category logic #}
```
Comment (Scotchester) [inline cfgov/jinja2/v1/_includes/organisms/post-preview.html]: Consider wording it more specifically like the suggestion below, and consider moving it inside the `if` statement two lines below.

```suggestion
                {# If rendering a blog post in the newsroom, use the 'Blog' aggregate category instead of the post's true category. #}
```

File: cfgov/jinja2/v1/_includes/organisms/post-preview.html [hunk 14]
```diff
@@ -77,6 +77,26 @@
...
+                    {{ category_slug.render(category='blog', href=url) }}
+                {% else %}
+                    {% for cat in post.categories.all() %}
+                        {% if loop.index > 1 %}
+                            <span class="m-meta-header_separator">|</span>
+                        {% endif %}
+                        <span class="m-meta-header_category">{{ category_slug.render(category=cat.name, href=url) }}</span>
+                    {% endfor %}
+                {% endif %}
+                {# Pipe ("|") separator logic. #}
```
Comment (Scotchester) [inline cfgov/jinja2/v1/_includes/organisms/post-preview.html]: This comment seems unnecessary.

```suggestion
```

File: cfgov/unprocessed/css/organisms/post-preview.less (hunk 12 above)
Comment (Scotchester) [reply cfgov/unprocessed/css/organisms/post-preview.less]: I agree that the `TODO` or issue would be helpful here.

File: cfgov/unprocessed/css/organisms/post-preview.less [hunk 15]
```diff
@@ -102,20 +135,48 @@
...
+        padding-right: unit( @grid_gutter-width / @base-font-size-px, em );
     }
 
     &_description {
         margin-bottom: unit( @grid_gutter-width / 2 / @base-font-size-px, em);
     }
 
     &_image-container {
+        grid-area: image;
+        width: unit( 270px / @base-font-size-px, em);
```
Comment (Scotchester) [inline cfgov/unprocessed/css/organisms/post-preview.less]: The `width` property and I assume also `grid-area` need to remain in a media query for images to be full-width on XS and S displays. A comment explaining the magic number would also be helpful.

```suggestion
        width: unit( 270px / @base-font-size-px, em );
```

File: cfgov/unprocessed/css/organisms/post-preview.less (hunk 12 above)
Comment (mistergone) [reply cfgov/unprocessed/css/organisms/post-preview.less]: I will create an issue about this `float`!
